pymlx/model.py

```diff
@@ -29,33 +32,47 @@ def get_feature_names(self):
         return self.featurizer.in_feature_names
 
     # ToDo: implement scoring for row
-    def predict(self, features):
+    def predict(self, features):  # <-- meets error
         """
         :param features: list or array of feature values.
         Missing values MUST be encoded as nan, not None.
         :return: probability of the instance classified as True
         """
-        indices, values = self.featurizer.featurize_row(features)
+
+        indices, values = self.featurizer.featurize_row(features) # <-- input features must be a Pandas Series
         features = csr_matrix((values, indices, [0, len(values)]),
                               shape=(1, self._num_features),
                               dtype=float32)
         return self.predictor.predict_proba(features)[0][1]
 
     def bulk_predict(self, test_data):
         if isinstance(test_data, DataFrame):
-            test_data = self.featurizer.transform(test_data)
-        assert isinstance(test_data, csr_matrix)
-        return self.predictor.predict_proba(test_data)[:, 1]
+            test_data = self.featurizer.transform(test_data, return_dataframe=True)
+        # assert isinstance(test_data, csr_matrix) # <--
+        result = self.predictor.predict_proba(test_data)
+        return result[:, 1], result
 
     # REVIEW: this is specific to xgboost, which is temporary
     # ToDo:
     #   - Absorb xgboost
     #   - Each predictor should indicate whether it supports features importance
+    #   11/09/2018: has just implement for xgboost and lightgbm
     def get_fscores(self):
-        feature_names = self.featurizer.out_feature_names
-        fscores = self.predictor.booster().get_fscore()
-        fscores = {feature_names[int(f[1:])]: fscores[f] for f in fscores}
-        return Series(fscores).sort_values(ascending=False)
+
+        if isinstance(self.predictor, XGBModel):
```

Comment:
It would be more robust to check based on `self.predictor.__module__` . That way, users don't have to install all libraries if they don't use many of them.

Also use 
```
if xgboost:
elif lightgbm
else:
    raise NotImplementedError("this predictor is not yet supported in this function")
```